File: 11713 Abstract Names/solution.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool isVowel(char ch) {
    return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
}

int main() {
	int n;
	scanf("%d", &n);
	char realName[21], gameName[21];
	while(n--) {
	    scanf("%s %s", realName, gameName);
	    int len = strlen(realName);
	    if(len == strlen(gameName)) {
	        bool same = true;
	        for(int i = 0; i < len && same; ++i) {
	            if(isVowel(realName[i]) != isVowel(gameName[i]) || (!isVowel(realName[i]) && realName[i] != gameName[i])) {
	                puts("No");
	                same = false;
	            }
	        }
	        if(same)
	            puts("Yes");
	    } else
	        puts("No");
	}
	return 0;
}

File: 11713 Abstract Names/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		st.nextToken();
		int n = (int)st.nval;
		StringBuilder output = new StringBuilder();
		while(n-- > 0) {
		    st.nextToken();
		    String realName = st.sval;
		    st.nextToken();
		    String gameName = st.sval;
		    if(realName.length() == gameName.length()) {
		        boolean same = true;
		        for(int i = 0; i < realName.length() && same; ++i) {
		            if(isVowel(realName.charAt(i)) != isVowel(gameName.charAt(i)) || (!isVowel(realName.charAt(i)) && realName.charAt(i) != gameName.charAt(i))) {
    	                output.append("No\n");
    	                same = false;
    	            }
		        }
		        if(same)
		            output.append("Yes\n");
		    } else
		        output.append("No\n");
		}
		System.out.print(output);
	}
	
	static boolean isVowel(char ch) {
	    return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
	}
}
